### DISCUSSION TITLE
spotbug vs error-prone

### DISCUSSION THREAD

**Discussion by GeorgeSalu** [spotbugs/spotbugs#2565] [General]
What is the difference between spotbugs and google error-prone? Are the detected bugs different??

**Comment by JuditKnoll** [comment]
These are two completely different Static Analysis tools for Java. There is an overlap between the detected bugs, but the bugcodes are not the same, and there may be edge cases even for one bugtype in which one tool finds a hit, but the other doesn't. There are several studies comparing static analysis tools including both spotbugs and error-prone.

**Comment by hazendaz** [comment]
Closing as answered, there are a lot of static analysis tools and even some like open rewrite that intend to not only report the issues but fix them through AI.  So in any case, use them all provided they don't overly slow your builds as there are always going to be things you find with one you won't find with others.
